src/backends/elasticsearch/Elasticsearch.ts

```diff
@@ -162,10 +165,18 @@ export class Elasticsearch implements IDataSource {
     search.sort = [
       {
         [this.fieldsConfig.timestamp]: {
-          order: (searchAfterAscending === true) ? 'asc' : 'desc'
+          order: (searchAfterAscending === true) ? 'asc' : 'desc',
+          numeric_type: "date"
         }
       }
     ]
+    // ES versions 7.7-7.9 have a bug when using date_nanos as a sort field. We can avoid this by forcing timestamps
+    // to have ms resolution instead, with the slight regression that logs may not have a stable sorting behavior
+    // (logs within 1ms will be considered as equal, and thus ordering dependent on aggregation.)
+    // see https://github.com/elastic/elasticsearch/issues/63719 for details
+    if (this.fieldsConfig.disableTimestampNano) {
```

Comment:
my thought here is that if you are adding secondary key, you explicitly _want_ nano precision (otherwise standard timestamp should be fine), so it should just always be enabled (or just don't use secondary key)

Reply:
(also, we don't technically require secondary index to even be a time, eg log counter?)

Reply:
Yeah, I guess let's hope someone doesn't enable a date_nanos secondary index